homeassistant/components/switchbee/__init__.py

```diff
@@ -0,0 +1,137 @@
+"""The SwitchBee Smart Home integration."""
+
+from __future__ import annotations
+
+import logging
+
+import switchbee
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, SCAN_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SwitchBee Smart Home from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    central_unit = entry.data[CONF_HOST]
+    user = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = switchbee.SwitchBee(central_unit, user, password, websession)
+    resp = await api.login()
+    if resp[switchbee.ATTR_STATUS] != switchbee.STATUS_OK:
+        _LOGGER.error(
+            "Failed to login to the central unit (%s) with the user %s %s",
+            central_unit,
+            central_unit,
+            resp,
+        )
```

Comment:
```suggestion
        _LOGGER.error(
            "Failed to login to the central unit (%s) with the user %s %s",
            central_unit,
            central_unit,
            resp,
        )
```
Please include this message in the `PlatformNotReady` instead of logging error. This will make sure it is only logged once.
e.g. `raise PlatformNotReady("some error")`